### PR TITLE
Fix build with Qt5.

### LINKED ISSUE #138: Error building cartographer_rviz after latest update
Hi
I updated cartographer_ros from a previous version (Oct 6, commit 1418902493115cad90d6ee93877c5a602a78614f) to the most up-to-date one (Oct 21, commit 20487028ed36c558336a873860cfe90340e8662f). While building completed without errors previously, I now get the following error:

> Errors     << cartographer_rviz:make /home/ros/catkin_ws/logs/cartographer_rviz/build.make.005.log                                      
In file included from /home/ros/catkin_ws/build/cartographer_rviz/cartographer_rviz/../../../src/overlay/cartographer_ros/cartographer_rv
iz/cartographer_rviz/drawable_submap.h:35:0,
                 from /home/ros/catkin_ws/build/cartographer_rviz/cartographer_rviz/moc_drawable_submap.cpp:9,
                 from /home/ros/catkin_ws/build/cartographer_rviz/cartographer_rviz/drawable_submap_automoc.cpp:2:
/opt/ros/kinetic/include/rviz/display_context.h:34:19: fatal error: QObject: No such file or directory
compilation terminated.
make[2]: *** [cartographer_rviz/CMakeFiles/drawable_submap.dir/drawable_submap_automoc.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /home/ros/catkin_ws/src/overlay/cartographer_ros/cartographer_rviz/cartographer_rviz/drawable_submap.h:35:0,
                 from /home/ros/catkin_ws/src/overlay/cartographer_ros/cartographer_rviz/cartographer_rviz/drawable_submap.cc:17:
/opt/ros/kinetic/include/rviz/display_context.h:34:19: fatal error: QObject: No such file or directory
compilation terminated.
make[2]: *** [cartographer_rviz/CMakeFiles/drawable_submap.dir/drawable_submap.cc.o] Error 1
make[1]: *** [cartographer_rviz/CMakeFiles/drawable_submap.dir/all] Error 2
make: *** [all] Error 2

My setup:
Ubuntu 16.04
ROS Kinetic
x86_64

I do realize that you mention ROS Kinetic not being fully supported by cartographer_ros, however as there were no build errors previously and now there are, I thought I'd report this anyway.

Regards

### PR DESCRIPTION
**Pull request by SirVer**
- Include Qt5 include directories into search path. Fixes #138.
- Supersedes and fixes #124.

### CODE REVIEW

File: scripts/install_debs.sh [hunk 1]
```diff
@@ -26,5 +26,6 @@ sudo apt-get update
 sudo apt-get install -y ninja-build
 
 # Install rosdep dependencies.
+rosdep init
```
Comment (damonkohler) [inline scripts/install_debs.sh]: This should probably not be done. It's already handled by the base image.

File: docs/source/index.rst [hunk 2]
```diff
@@ -65,6 +65,7 @@ We recommend using `wstool <http://wiki.ros.org/wstool>`_ and `rosdep
     wstool update -t src
 
     # Install deb dependencies.
+    rosdep init
```
Comment (ojura) [inline docs/source/index.rst]: http://wiki.ros.org/rosdep says this should be invoked with sudo

File: cartographer_rviz/CMakeLists.txt [hunk 3]
```diff
@@ -40,12 +40,15 @@ set(CMAKE_AUTOMOC ON)
 if(rviz_QT_VERSION VERSION_LESS "5")
   message(STATUS "Using Qt4 based on the rviz_QT_VERSION: ${rviz_QT_VERSION}")
   find_package(Qt4 ${rviz_QT_VERSION} EXACT REQUIRED QtCore QtGui)
+  include_directories(${Qt4_INCLUDE_DIRS})
   include(${QT_USE_FILE})
 else()
   message(STATUS "Using Qt5 based on the rviz_QT_VERSION: ${rviz_QT_VERSION}")
   find_package(Qt5 ${rviz_QT_VERSION} EXACT REQUIRED Core Widgets)
   find_package(Qt5Widgets REQUIRED)
```
Comment (ojura) [inline cartographer_rviz/CMakeLists.txt]: I believe this line is redundant, this is performed by the line above.

File: cartographer_rviz/CMakeLists.txt [hunk 4]
```diff
@@ -40,12 +40,15 @@ set(CMAKE_AUTOMOC ON)
...
   set(QT_LIBRARIES Qt5::Widgets)
+  include_directories(${Qt5Widgets_INCLUDE_DIRS})
+  include_directories(${Qt5_INCLUDE_DIRS})
```
Comment (ojura) [inline cartographer_rviz/CMakeLists.txt]: This line too, Qt5_INCLUDE_DIRS is not defined.

**Pull request merged**